src/main/java/com/google/cloud/solutions/spannerddl/diff/DdlDiff.java

```diff
@@ -707,6 +707,11 @@ private static String getDatabaseNameFromAlterDatabase(List<ASTddl_statement> st
    * @throws DdlDiffException if there is an error in parsing the DDL
    */
   public static List<ASTddl_statement> parseDdl(String original) throws DdlDiffException {
+    // the annotations are prefixed with "--" so that SQL file remains valid.
+    // strip the comment prefix before so that annotations can be parsed.
+    // otherwise they will be ignored as comment lines
+    original = original.replaceAll("-- *@", "@");
```

Comment:
There is a risk here of an expression "-- @" appearing in static strings being matched. 
This would be unusual, but would cause confusing parsing errors to users who don't use these type of annotations

If your annotation comments always appear on single lines without anything before them on the same line, then maybe the following replacement might be better: 

```
original = Pattern.compile("^\s*--\s+@", Pattern.MULTILINE).matcher(original).replaceAll("@")
```

(I know the same applies for the replacements of comments below as well, and I should fix that somehow...)

Reply:
Can you also add a command line option to enable annotation processing and handling (defaulting to off) - this is a non-standard spanner feature, and may cause issues for other users. 

The easiest way of implementing it would be to only make this replacement if the option is enabled - if it is disabled any annotations would be removed as if they were comments.